src/Core/src/Platform/Windows/WindowExtensions.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace Microsoft.Maui
+{
+	public static class WindowExtensions
+	{
+		public static void UpdateTitle(this UI.Xaml.Window nativeWindow, IWindow window) =>
+			nativeWindow.Title = window.Title;
```

Comment:
Windows is the only platform right now that supports arbitrary titles independent of pages.